consumer_group.go

```diff
@@ -433,6 +434,13 @@ type ConsumerGroupSession interface {
 	// allows incrementing the offset. cf MarkOffset for more details.
 	ResetOffset(topic string, partition int32, offset int64, metadata string)
 
+	// NextOffset returns the next offset that should be consumed for the managed
+	// partition, accompanied by metadata which can be used to reconstruct the state
+	// of the partition consumer when it resumes. NextOffset() will return
+	// `config.Consumer.Offsets.Initial` and an empty metadata string if no offset
+	// was committed for this partition yet.
+	NextOffset(topic string, partition int32) (int64, string, error)
```

Comment:
I don't really see how this is useful.

Reply:
well, what I'm essentially doing is trying to create an exactly once semantics for uploading  kafka messages to a bucket (inspired heavily by [secor](https://github.com/pinterest/secor)).
In order to do so i'm keeping track of the current offsets, and in case of rebalance, if the same partition is not being managed by the same client, i'm deleting the messages locally.
Otherwise, in order to minimise processing the same message twice or more, I continue from the same location 
In order to do so, I really benefit from knowing which offset in the next one for the current partition